src/block_pos.rs

```diff
@@ -23,6 +24,28 @@ impl BlockPos {
     pub fn at(pos: impl Into<Vec3<f64>>) -> Self {
         pos.into().floor().as_::<i32>().into()
     }
+
+    /// Get a new [`BlockPos`] that is a adjacent to this position in `dir`
+    /// direction.
+    ///
+    /// ```rust
+    /// use valence::block::BlockPos;
+    /// use valence::client::BlockFace;
+    ///
+    /// let pos = BlockPos::new(0, 0, 0);
+    /// let adj = pos.get_in_direction(BlockFace::South);
+    /// assert_eq!(adj, BlockPos::new(0, 0, 1));
+    /// ```
+    pub fn get_in_direction(&self, dir: BlockFace) -> BlockPos {
```

Comment:
Should take `self` by value because `BlockPos` is `Copy`.
```suggestion
    pub fn get_in_direction(self, dir: BlockFace) -> BlockPos {
```